src/Sulu/Bundle/PreviewBundle/Resources/js/views/Preview/Preview.js

```diff
@@ -0,0 +1,265 @@
+// @flow
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {action, autorun, observable, toJS} from 'mobx';
+import {observer} from 'mobx-react';
+import debounce from 'debounce';
+import Router from 'sulu-admin-bundle/services/Router';
+import {sidebarStore} from 'sulu-admin-bundle/containers/Sidebar';
+import {FormStore} from 'sulu-admin-bundle/containers/Form';
+import ResourceStore from 'sulu-admin-bundle/stores/ResourceStore';
+import Requester from 'sulu-admin-bundle/services/Requester';
+import Toolbar from 'sulu-admin-bundle/components/Toolbar';
+import Loader from 'sulu-admin-bundle/components/Loader';
+import previewStyles from './preview.scss';
+import previewConfigStore from './stores/PreviewConfigStore';
+
+type Props = {
+    resourceStore: ResourceStore,
+    formStore: FormStore,
+    router: Router,
+};
+
+@observer
+export default class Preview extends React.Component<Props> {
+    @observable iframeRef: ?Object;
+    @observable token: ?string;
+    @observable started: boolean = false;
+
+    typeDisposer: () => void;
+    dataDisposer: () => void;
+
+    componentDidMount() {
+        if (previewConfigStore.mode === 'auto') {
+            this.startPreview();
+        }
+    }
+
+    @action startPreview = () => {
+        const {
+            resourceStore,
+            formStore,
+            router: {
+                attributes: {
+                    locale,
+                },
+            },
+        } = this.props;
+
+        const route = previewConfigStore.generateRoute('start', {
```

Comment:
Maybe this should be a `preview` service? `generateRoute` does not sound like a function that should be in a config object.

Reply:
The the `preview` service could have a `start` method taking these parameters, and all the `Requester` instances could probably be gone from this component.